packages/grid/_modules_/grid/hooks/features/filter/useGridFilter.ts

```diff
@@ -271,6 +271,8 @@ export const useGridFilter = (apiRef: GridApiRef, rowsProp: GridRowsProp): void
     [apiRef],
   );
 
+  const getVisibleRowModels = React.useCallback(() => visibleRowModels, [visibleRowModels]);
```

Comment:
Did you consider only have a dependency on the `apiRef` and read from the state? It would lead to fewer changes of the `getVisibleRowModels` API method. On the other hand, it might also lead to making `getVisibleRowModels ` return a state that isn't yet rendered. I wonder.